firebase/functions/src/index.ts

```diff
@@ -0,0 +1,532 @@
+            // aren't properly defined
+            Object.defineProperty(zone.cards, id, {
+                value: {
+                    id: id,
+                    scryfall_id: card.scryfall_id,
+                    state: {
+                        owner: player,
+                        controller: player,
+                        zone: Zone.Library,
+                        position: position,
+                        tapped: false,
+                        face_up: false,
+                        clone_of: null,
+                        is_morph: false,
+                        is_token: false,
+                        power: null,
+                        toughness: null,
+                        attachments: {
+                            counters: {
+
+                            },
+                            permanents: {
+
+                            }
+                        }
+                    }
+                },
+                enumerable: true
+            });
+            position++;
+        }
+    });
+    
+    return zone;
+};
+
+const createCommandZone = function(params) {
+    const player = params.player;
+    const deck = params.deck;
+    const zone = createEmptyZone();
+
+    Object.keys(deck.cards).forEach(key => {
+        const card = deck.cards[key];
+        if (card.board === Board.Command) {
+            const id = uuidv4();
+            Object.defineProperty(zone.cards, id, {
+                value: {
+                    id: id,
+                    scryfall_id: card.scryfall_id,
+                    state: {
+                        owner: player,
+                        controller: player,
+                        zone: Zone.Command,
+                        position: 0,
+                        tapped: false,
+                        face_up: true,
+                        clone_of: null,
+                        is_morph: false,
+                        is_token: false,
+                        power: null,
+                        toughness: null,
+                        attachments: {
+                            counters: {
+
+                            },
+                            permanents: {
+
+                            }
+                        }
+                    }
+                },
+                enumerable: true
+            });
+        }
+    });
+    return zone;
+}
+
+const regenerateObject = function(obj: any): any {
+    return JSON.parse(JSON.stringify(obj));
+}
+
+const lookupDeck = async function(player: string, deckId: string): Promise<Firestore.DocumentSnapshot> {
+    let querySnapshot;
+    try {
+        querySnapshot = await firestore.collection('Users')
+            .where('username', '==', player)
+            .limit(1)
+            .get();
+    } catch (e) {
+        console.error('Caught error searching for user', e);
+        throw createError(500);
+    }
+    
+    if (querySnapshot.docs.length === 0) {
+        throw createError(400, null, {
```

Comment:
I had debated between 400 and 404 at first, but since these HTTP calls are POST requests, it seemed like 400 made more sense. I'm not sure what is de facto behavior though.

Reply:
Ahhh, I didn't realize these were all POST requests. In that case, it makes sense